### PR TITLE
GitHub+Travis: avoid Docker Hub pull limit by logging into docker account

### PR DESCRIPTION
**Pull request by guggero**
There is a new image pull limit enforced by Docker Hub for anonymous
users. Since we seem to be sharing that pull limit with other jobs
running from the same IP, we want to log in to get our own, exclusive
pull limit of currently 200 pulls per 6 hours.

@Roasbeef: Need new secrets in Travis:
- `DOCKER_USERNAME`: same value as on GitHub
- `DOCKER_PASSWORD` same value as `DOCKER_API_KEY` on GitHub

### CODE REVIEW

**Comment by guggero** [comment]
Hmm... The action seems to fail. @Roasbeef are the GH secrets limited to certain workflows? If yes, we might need to add the main CI workflow to the allow list.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -53,9 +53,18 @@ jobs:
         with:
           go-version: '${{ env.GO_VERSION }}'
 
+      # To avoid sharing the per-IP pull limit on Docker Hub for unauthenticated
+      # users with other jobs running on the same VM we log in to get our own,
+      # currently 200 pulls per 6 hours limit.
+      - name: Login to DockerHub
+        uses: lightninglabs/gh-actions/login-action@2021.01.25.00
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
```
[the remark this answers is not in the document]
Comment (guggero) [reply .github/workflows/main.yml:62]: Actually, that's a great idea!

**Comment by AaronDewes** [comment]
> Are the GH secrets limited to certain workflows?

I think they don't work on PRs, because PRs don't have access to secrets (for security purposes, if anyone modifies the workflow, secrets can be exposed).

**Comment by guggero** [comment]
It seems like this is no longer needed. Didn't happen again AFAIK.

**Pull request closed without merging**